File: Instend.API/Client/src/api/NewsController.js
import { instance } from "../state/application/Interceptors";
import NewsState from "../state/entities/NewsState";

class NewsController {
    static SetNewsRequest = async (lastPublicationDate = '') => {
        await instance
            .get(`api/news?lastPublicationDate=${lastPublicationDate}`)
            .then((response) => {
                if (!response || !response.data || !response.data.length) {
                    NewsState.setHasMoreState(false);
                    return;
                }

                if (response.data && response.data.length && response.data.length > 0) {
                    NewsState.addNews(response.data);
                }

                NewsState.setHasMoreState(response.data.length >= 5);
            })
            .catch(e => {
                console.error(e);
            });
    }

    static GetAccountPublications = async (accountId, lastPublicationDate = '', setPublications, setHasMoreState) => {
        await instance
            .get(`api/account/publications?lastPublicationDate=${lastPublicationDate}&accountId=${accountId}`)
            .then((response) => {
                if (!response || !response.data || !response.data.length) {
                    setHasMoreState(false);
                    return;
                }

                if (response.data && response.data.length && response.data.length > 0) {
                    setPublications(prev => [...prev, ...response.data]);
                }

                setHasMoreState(response.data.length >= 5);
            })
            .catch(e => {
                console.error(e);
            });
    }
}

export default NewsController;

File: Instend.API/Server/Controllers/Publications/NewsController.cs
using Instend.Core.Dependencies.Repositories.Account;
using Instend.Core.Dependencies.Services.Internal.Helpers;
using Instend.Services.Internal.Handlers;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Instend_Version_2._0._0.Server.Controllers.Comments
{
    [ApiController]
    [Route("[controller]")]
    public class NewsController : ControllerBase
    {
        private readonly IRequestHandler _requestHandler;

        private readonly FilesController _accountsRepository;

        private readonly IPublicationsRepository _publicationsRepository;

        private readonly ISerializationHelper _serializationHelper;

        public NewsController
        (
            IRequestHandler requestHandler, 
            FilesController accountsRepository,
            IPublicationsRepository publicationsRepository,
            ISerializationHelper serializationHelper
        )
        {
            _requestHandler = requestHandler;
            _accountsRepository = accountsRepository;
            _publicationsRepository = publicationsRepository;
            _serializationHelper = serializationHelper;
        }

        [HttpGet]
        [Authorize]
        [Route("/api/news")]
        public async Task<IActionResult> GetPublictions(string? lastPublicationDate)
        {
            DateTime date;

            if (lastPublicationDate == null)
            {
                date = DateTime.Now;
            }
            else if (!DateTime.TryParse(lastPublicationDate, out date))
            {
                return BadRequest("Invalid date format");
            }

            var userId = _requestHandler.GetUserId(Request.Headers["Authorization"]);

            if (userId.IsFailure)
                return BadRequest(userId.Error);

            var account = await _accountsRepository
                .GetByIdAsync(Guid.Parse(userId.Value));

            if (account == null)
                return Conflict("Account not finded");

            var publications = await _publicationsRepository
                .GetNewsByAccount(date, [account.Id, ..account.Following.Select(f => f.AccountId)], account.Id, 5);

            return Ok(_serializationHelper.SerializeWithCamelCase(publications));
        }

        [HttpGet]
        [Authorize]
        [Route("/api/account/publications")]
        public async Task<IActionResult> GetPublictions(string? lastPublicationDate, Guid accountId)
        {
            DateTime date;

            if (lastPublicationDate == null)
            {
                date = DateTime.Now;
            }
            else if (!DateTime.TryParse(lastPublicationDate, out date))
            {
                return BadRequest("Invalid date format");
            }

            var publications = await _publicationsRepository
                .GetAccountPublications(accountId, date, 5);

            return Ok(_serializationHelper.SerializeWithCamelCase(publications));
        }
    }
}